### PR TITLE
Adopt copier template

### PR DESCRIPTION
**Pull request by evalott100**
Copier template adoption. If we decide to adopt pyright then I can do that in another PR.

### CODE REVIEW

**Comment by evalott100** [comment]
Still to adjust `__init__` to be ruff compliant. This is a much larger issue :sweat_smile:

**Comment by evalott100** [comment]
That was much easier than I thought it would be!

File: README.md [hunk 1]
```diff
@@ -1,25 +1,26 @@
-:html_theme.sidebar_secondary.remove:
+<img src="https://raw.githubusercontent.com/bluesky/event-model/refs/heads/main/docs/images/event-model-logo.svg"
+     style="background: none" width="120px" height="120px" align="center">
 
-.. include:: ../README.rst
-    :end-before: when included in index.rst
+Data model used by the bluesky ecosystem.
```
Comment (coretl) [inline README.md:4]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
 
-===================================
- Bluesky Event Model Documentation
-===================================
+[![CI](https://github.com/bluesky/event-model/actions/workflows/ci.yml/badge.svg)](https://github.com/bluesky/event-model/actions/workflows/ci.yml)
+[![Coverage](https://codecov.io/gh/bluesky/event-model/branch/main/graph/badge.svg)](https://codecov.io/gh/bluesky/event-model)
+[![PyPI](https://img.shields.io/pypi/v/event-model.svg)](https://pypi.org/project/event-model)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+# Event Model
```
Comment (coretl) [inline README.md:11]:
```suggestion
Data model used by the bluesky ecosystem.

# Event Model
```

**Comment by danielballan** [comment]
If I don't review this by end of Wednesday, go ahead without me on Thursday.

File: docs/conf.py [hunk 3]
```diff
@@ -54,17 +51,22 @@
     "sphinx_copybutton",
     # For the card element
     "sphinx_design",
+    "sphinx_design",
```
Comment (danielballan) [inline docs/conf.py:54]: Duplicate.

File: src/event_model/tests/test_run_router.py [hunk 4]
```diff
@@ -340,7 +340,7 @@ def exception_callback_factory(start_doc_name, start_doc):
     rr("start", start_doc)
 
     descriptor_doc = {"run_start": "abcdef", "uid": "ghijkl"}
-    with pytest.raises(Exception):
+    with pytest.raises(UserWarning):
```
Comment (danielballan) [inline src/event_model/tests/test_run_router.py:343]: Why the change? If this changed to a warning shouldn't the test use `pytest.warns`?

Comment (evalott100) [reply src/event_model/tests/test_run_router.py]: The change was just because linting wasn't a fan of`pytest.raises` with a base `Exception`, but `rt("descriptor", descriptor_doc)` both raises `Exception` and `warns` so the `pytest.raises(UserWarning)` erroneously caught it.

**Review by coretl — APPROVED** [review]
